File: foodwithfriends-trigger/src/main/java/de/koanam/foodwithfriends/trigger/model/entity/Date.java
package de.koanam.foodwithfriends.trigger.model.entity;

import java.time.LocalDateTime;
import java.util.List;

import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.JoinTable;
import javax.persistence.ManyToMany;
import javax.persistence.OneToMany;

@Entity
public class Date {

	@Id
	private Long id;
	
	private LocalDateTime datetime;
	
	@ManyToMany
	@JoinTable(name ="user_date", inverseJoinColumns = @JoinColumn(name = "user_id", table="user"))
	private List<User> users;
	
	@OneToMany
	private List<Vote> votes;

	public Date() {
	}

	public Date(List<User> users) {
		this.users = users;
	}

	public List<User> getUsers() {
		return users;
	}

	public void setUsers(List<User> users) {
		this.users = users;
	}

	public LocalDateTime getDatetime() {
		return datetime;
	}

	public void setDatetime(LocalDateTime datetime) {
		this.datetime = datetime;
	}
	
		
}

File: foodwithfriends/lib/model/Date.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:foodwithfriends/model/User.dart';
import 'package:foodwithfriends/model/Vote.dart';

part 'Date.g.dart';

@JsonSerializable()
class Date{

  int id;
  @JsonKey(name: "datetime")
  DateTime datetime;
  @JsonKey(name: "user_dates")
  List<User> users;
  List<Vote> votes;

  Date();

  factory Date.fromJson(Map<String, dynamic> json) => _$DateFromJson(json);

  Map<String, dynamic> toJson() => _$DateToJson(this);
}